aineko/core/dataset.py

```diff
@@ -132,30 +140,99 @@ def _validate_message(
             message = message.decode("utf-8")
         return json.loads(message)
 
+    def _update_offset_to_latest(self) -> None:
+        """Updates offset to latest.
+
+        Note that the initial call, for this method might take
+        a while due to consumer initialization.
+        """
+        partitions = self.consumer.assignment()
+        # Initialize consumers if not already initialized by polling
+        while not partitions:
+            self.consumer.poll(timeout=0)
+            partitions = self.consumer.assignment()
+
+        for partition in partitions:
+            partition.offset = (
+                self.consumer.get_watermark_offsets(
+                    partition, cached=self.cached
+                )[1]
+                - 1
+            )
+
+        self.consumer.assign(partitions)
+
     def consume(
         self,
-        how: str = "next",
-        timeout: Optional[int] = None,
+        how: Literal["next", "last"] = "next",
+        timeout: Optional[float] = None,
     ) -> Optional[dict]:
-        """Reads a message from the dataset.
+        """Polls a message from the dataset.
 
         Args:
-            how: how to read the message
+            how: how to read the message.
                 "next": read the next message in the queue
+                "last": read the last message in the queue
+            timeout: seconds to poll for a resopnse from kafka broker.
+            If using how="last",
```

Comment:
```suggestion
            timeout: seconds to poll for a response from kafka broker when using
                how="last"
```
typo + clearer description imo